### PR TITLE
Add new Psalm LSP linter

### PR DESCRIPTION
**Pull request by muglug**
Replacement for https://github.com/w0rp/ale/pull/2008

Psalm was just updated to support the Language Server Protocol.

This PR adds a `psalm_lsp` plugin that uses LSP, and marks the existing `psalm` plugin as deprecated.

cc @twitchard

### CODE REVIEW

**Comment by RyanSquared** [comment]
Not sure why CI is taking so long here, it's been well over an hour

**Comment by RyanSquared** [comment]
Can you reorder the table of contents to match up properly?

```diff
--- /tmp/table-of-contents.jA3MVkT1	2018-10-23 14:16:44.573557806 +0000
+++ /tmp/headings.vhbc7tMr	2018-10-23 14:16:44.597570097 +0000
@@ -194,8 +194,8 @@
   phpcs, ale-php-phpcs
   phpmd, ale-php-phpmd
   phpstan, ale-php-phpstan
-  psalm, ale-php-psalm
   psalm_lsp, ale-php-psalm_lsp
+  psalm, ale-php-psalm
   php-cs-fixer, ale-php-php-cs-fixer
 po, ale-po-options
   write-good, ale-po-write-good
```

File: doc/ale-php.txt [hunk 1]
```diff
@@ -169,6 +169,23 @@ g:ale_php_phpstan_configuration               *g:ale_php_phpstan_configuration*
...
+  Default: `'psalm'`
+
+  This variable sets the executable used for Psalm's language server.
+
+g:ale_php_psalm_lsp_use_global                 *g:ale_php_psalm_lsp_use_global*
+                                               *b:ale_php_psalm_lsp_use_global*
+  Type: |Boolean|
+  Default: `get(g:, 'ale_use_global_executables', 0)`
+
+  This variable force globally installed Psalm language server.
```
Comment (RyanSquared) [inline doc/ale-php.txt]: Nitpick: `s/force/forces/`

Comment (muglug) [reply doc/ale-php.txt]: Fixed both here and the place I blindly copy-pasted from

**Comment by muglug** [comment]
Everything should now be good!

**Comment by RyanSquared** [comment]
:+1: from me, but I'll let w0rp do a review as well in case I miss something.

**Comment by RyanSquared** [comment]
Many people just pull master. If it was merged, it's incompatible.

**Comment by w0rp** [comment]
In case others are wondering, the CI took a long time a few days ago because GitHub itself was having some problems.

**Comment by w0rp** [comment]
@muglug I think you are right. I didn't release that the `psalm` linter was added so recently. I'll merge the previous pull request. I think it's fine to change something like that if it hasn't been tagged in a release yet.

**Comment by w0rp** [comment]
@muglug You can open up a PR which changes the linter to just be an LSP linter instead again, and we can merge that if everything is okay there. Sorry for the confusion.

**Comment by muglug** [comment]
Thanks, I've created one

**Pull request closed without merging**